packages/@uppy/dashboard/src/components/AddFiles.js

```diff
@@ -1,101 +1,113 @@
-const ActionBrowseTagline = require('./ActionBrowseTagline')
 const { localIcon } = require('./icons')
 const { h, Component } = require('preact')
 
-const poweredByUppy = (props) => {
-  return <a tabindex="-1" href="https://uppy.io" rel="noreferrer noopener" target="_blank" class="uppy-Dashboard-poweredBy">{props.i18n('poweredBy')} <svg aria-hidden="true" class="UppyIcon uppy-Dashboard-poweredByIcon" width="11" height="11" viewBox="0 0 11 11">
-    <path d="M7.365 10.5l-.01-4.045h2.612L5.5.806l-4.467 5.65h2.604l.01 4.044h3.718z" fill-rule="evenodd" />
-  </svg><span class="uppy-Dashboard-poweredByUppy">Uppy</span></a>
-}
 class AddFiles extends Component {
   constructor (props) {
     super(props)
-    this.handleClick = this.handleClick.bind(this)
+
+    this.triggerFileInputClick = this.triggerFileInputClick.bind(this)
+
+    this.renderPoweredByUppy = this.renderPoweredByUppy.bind(this)
+    this.renderHiddenFileInput = this.renderHiddenFileInput.bind(this)
+    this.renderDropPasteBrowseTagline = this.renderDropPasteBrowseTagline.bind(this)
+    this.renderMyDeviceAcquirer = this.renderMyDeviceAcquirer.bind(this)
+    this.renderAcquirer = this.renderAcquirer.bind(this)
   }
 
-  handleClick (ev) {
-    this.input.click()
+  triggerFileInputClick () {
+    this.fileInput.click()
   }
 
-  render () {
-    const hasAcquirers = this.props.acquirers.length !== 0
+  renderPoweredByUppy () {
+    return <a tabindex="-1" href="https://uppy.io" rel="noreferrer noopener" target="_blank" class="uppy-Dashboard-poweredBy">
+      {this.props.i18n('poweredBy')}
+      <svg aria-hidden="true" class="UppyIcon uppy-Dashboard-poweredByIcon" width="11" height="11" viewBox="0 0 11 11">
+        <path d="M7.365 10.5l-.01-4.045h2.612L5.5.806l-4.467 5.65h2.604l.01 4.044h3.718z" fill-rule="evenodd" />
+      </svg>
+      <span class="uppy-Dashboard-poweredByUppy">Uppy</span>
+    </a>
+  }
 
-    if (!hasAcquirers) {
-      return (
-        <div class="uppy-DashboardAddFiles">
-          <div class="uppy-DashboardTabs">
-            <ActionBrowseTagline
-              acquirers={this.props.acquirers}
-              handleInputChange={this.props.handleInputChange}
-              i18n={this.props.i18n}
-              i18nArray={this.props.i18nArray}
-              allowedFileTypes={this.props.allowedFileTypes}
-              maxNumberOfFiles={this.props.maxNumberOfFiles}
-            />
-          </div>
-          <div class="uppy-DashboardAddFiles-info">
-            { this.props.note && <div class="uppy-Dashboard-note">{this.props.note}</div> }
-            { this.props.proudlyDisplayPoweredByUppy && poweredByUppy(this.props) }
-          </div>
-        </div>
-      )
-    }
+  renderHiddenFileInput () {
+    return <input class="uppy-Dashboard-input"
+      hidden
+      aria-hidden="true"
+      tabindex={-1}
+      type="file"
+      name="files[]"
+      multiple={this.props.maxNumberOfFiles !== 1}
+      // Clear file input
```

Comment:
Maybe add that setting an empty `value=""` wasn’t enough, and we are doing this to fix `onBeforeFileAdded`? Not sure, but might be helpful in the future?

Reply:
@arturi, agreed, I'll add it.